mozilla-release/docshell/base/nsIWebNavigation.idl

```diff
@@ -279,14 +279,17 @@ interface nsIWebNavigation : nsISupports
    *        that at present this argument is only used with view-source aURIs
    *        and cannot be used to resolve aURI.
    *        This parameter is optional and may be null.
+   * @param aEnsurePrivate
+   *        Set whether to switch into private mode before loading.
```

Comment:
This flag we require in order to correctly drag tabs from private to normal windows.
The reason is that DocShell swapping between `xul:browser`s is only possible when their OriginAttributes match, including privateness. Thus, when creating a tab in a normal window to swap its DocShell with one of the private tab, we load it private.